modules/integration/tests-common/admin-clients/src/main/java/org/wso2/bps/integration/common/clients/bpmn/ActivitiRestClient.java

```diff
@@ -0,0 +1,736 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.bps.integration.common.clients.bpmn;
+
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpResponse;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.methods.*;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.apache.commons.logging.Log;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * This Class contains methods which uses the BPMN Rest Services to carryout BPMN Task
+ *
+ * @author WSO2
+ * @version 1.0
+ */
+public class ActivitiRestClient {
+
+	private static final Log log = LogFactory.getLog(ActivitiRestClient.class);
+	private final static String username = "admin";
+	private final static String password = "admin";
+	private final static String userClaim = "paul";
+	private final static String userDelegate = "will";
+	private int port;
+	private String hostname = "";
+	private String serviceURL = "";
+
+
+	public ActivitiRestClient(String portM, String hostnameM){
+		port = Integer.parseInt(portM);
+		hostname = hostnameM;
+		serviceURL = "http://" + hostnameM +":" + portM +"/bpmnrest/";
+	}
+
+	/**
+	 * Used put the thread in sleep until the tasks are generated
+	 */
+	public void waitForTaskGeneration() throws InterruptedException{
+		try {
+			Thread.sleep(1000);
+		}
+		catch (InterruptedException ie) {
+			log.error(" Thread Interuppted", ie);
+		}
+	}
+
```

Comment:
Remove the comment //region DeploymentMethods since there is a proper documentation comment now.